my-app/src/App.js

```diff
@@ -44,8 +50,35 @@ class App extends Component {
     this.addThing = this.addThing.bind(this);
     this.setPerson = this.setPerson.bind(this);
     this.setTag = this.setTag.bind(this);
+    this.addPerson = this.addPerson.bind(this);
+    this.addTagToPerson = this.addTagToPerson.bind(this);
+  }
+
+  addPerson = name => {
+    DataStore.addPersonByName(name)
+      .then((id)=>{
+        this.props.history.push('/person/'+id);
```

Comment:
I'm not familiar with this pattern of pushing things onto the history.  I'd expect the ReactRouter to provide some higher level facility for setting the "route" abstractly that would let you set variables like personId and it would compose it into the URL according to some pattern (that is also used for extracting personId from the URL).  I am willing to be wrong on this one, since I'm just carrying expectations from the angular router I've used in another project.